app/util/util.go

```diff
@@ -68,3 +69,34 @@ func LockHashToBytes(lockHash string) ([]byte, error) {
 func K1SignatureToBytes(signature string) ([]byte, error) {
 	return from0x(signature, k1SignatureLen)
 }
+
+// VCSInfoMap gets vcs information from bi and returns them as a map[string]string.
+func VCSInfoMap(bi *debug.BuildInfo) map[string]string {
+	ret := map[string]string{
+		"vcs.revision": "",
+		"vcs.time":     "",
+		"vcs.modified": "",
+	}
+
+	for _, element := range bi.Settings {
+		if _, ok := ret[element.Key]; ok {
+			ret[element.Key] = element.Value
+		}
+	}
+
+	return ret
+}
+
+// GitHash returns the git hash of the binary at the moment it was compiled.
+func GitHash() string {
+	raw, _ := debug.ReadBuildInfo()
+
+	info := VCSInfoMap(raw)
+
+	hash, found := info["vcs.revision"]
+	if !found {
+		panic("could not read git rev from binary, fatal!")
```

Comment:
This has gained a panic in the move, is it serious enough that we want to do that? Once anyone can compile from source and this auto sets and doesn't need custom build args its probably fine.

Reply:
Consider that those tags are embedded in the binary from the Go compiler itself. 

When you run “go build” from the cloned repository, it’ll get embedded automatically. Unless you clone, delete “.git” and then run build, this will never panic.

Reply:
Okay perfect, that was what I wanted to confirm